src/Microsoft.DotNet.Build.Tasks/GenerateTestExecutionScripts.cs

```diff
@@ -109,29 +112,25 @@ private void WriteCmdExecutionScript(string cmdExecutionTemplate, string outputP
             StringBuilder copyCommands = new StringBuilder();
             foreach (ITaskItem dependency in TestDependencies)
             {
-                string relativePath = dependency.GetMetadata("PackageRelativePath");
+                string relativeDestinationPath = dependency.GetMetadata("RelativeDestinationPath");
+                string packageRelativePath = dependency.GetMetadata("PackageRelativePath");
                 bool? useAbsolutePath = dependency.GetMetadata("UseAbsolutePath")?.Equals("true", StringComparison.OrdinalIgnoreCase);
                 if (useAbsolutePath == true)
                 {
                     string fullPath = dependency.GetMetadata("SourcePath");
                     fullPath = fullPath.Replace('/', '\\');
-                    copyCommands.AppendLine($"call :copyandcheck \"{fullPath}\" \"%EXECUTION_DIR%/{Path.GetFileName(fullPath)}\" || exit /b -1");
+                    copyCommands.AppendLine($"call :copyandcheck \"{fullPath}\" \"%EXECUTION_DIR%/{relativeDestinationPath}\" || exit /b -1");
                 }
                 // Generally anything without the relative path is just the test DLL and its directly referenced dependencies.  
                 // Every test project comes with 4 of them, so not producing a warning here.
-                else if (!string.IsNullOrEmpty(relativePath))
+                else if (!string.IsNullOrEmpty(packageRelativePath))
                 {
                     bool? preserveSubDirectories = dependency.GetMetadata("PreserveSubDirectories")?.Equals("true", StringComparison.OrdinalIgnoreCase);
-                    string filePath = Path.GetFileName(relativePath);
                     if (preserveSubDirectories == true)
                     {
-                        // PackageRelativePath contains (PackageName/VersionNumber/Directories/FileName). This is to remove the first two directories on 
-                        // the path to preserve just the directory structure.
-                        int indexOfSubDirectories = relativePath.IndexOf("\\", relativePath.IndexOf("\\")+1);
-                        filePath = relativePath.Substring(indexOfSubDirectories);
-                        copyCommands.AppendLine($"call :makedir \"%EXECUTION_DIR%{Path.GetDirectoryName(filePath)}\" ||  exit /b -1");
+                        copyCommands.AppendLine($"call :makedir \"%EXECUTION_DIR%{Path.GetDirectoryName(relativeDestinationPath)}\" ||  exit /b -1");
```

Comment:
@MattGal Fixed. I also added logic so that it only adds "makedir" commands if the directory to create is not just the EXECUTION_DIR.  Previously, it was adding this line several times, which is unnecessary:
`call :makedir "%EXECUTION_DIR%\" ||  exit /b -1`